client/auth.go

```diff
@@ -20,7 +18,7 @@ func WithDeviceID(deviceID string) LoginOpt {
 }
 
 // LoginUser will log in to a homeserver and create a new device on an existing user.
-func (c *CSAPI) LoginUser(t *testing.T, localpart, password string, opts ...LoginOpt) (userID, accessToken, deviceID string) {
+func (c *CSAPI) LoginUser(t TestLike, localpart, password string, opts ...LoginOpt) (userID, accessToken, deviceID string) {
```

Comment:
Does [testing.TB](https://pkg.go.dev/testing#TB) not suffice?

Reply:
That has too many useless functions we don't use like `Setenv(key, value [string](https://pkg.go.dev/builtin#string))`.